agent.go

```diff
@@ -480,6 +488,69 @@ func (p *pod) closeChannels() {
 	}
 }
 
+// listenToUdevEvents listens to udev events to detect device nodes added
+// and binds the device nodes to the /dev of all the containers.
+// This is required as a container has its /dev mounted with tmpfs with an initial
+// list of devices.
+func (p *pod) listenToUdevEvents(done chan struct{}) {
+	u := goudev.Udev{}
+
+	// Create a monitor listening on a NetLink socket.
+	monitor := u.NewMonitorFromNetlink("udev")
+
+	// Start monitor goroutine.
+	ch, _ := monitor.DeviceChan(done)
+
+	go func() {
+		fieldLogger := agentLog.WithField("subsystem", "udevlistener")
+
+		fieldLogger.Info("Started listening for all udev events")
+
+		for d := range ch {
+			fieldLogger.WithFields(logrus.Fields{
+				"udev-path":  d.Syspath(),
+				"udev-event": d.Action(),
+			}).Info("Received udev event")
+
+			// Ignore udev events for block devices, these are handled by another go-routine
+			if d.Subsystem() == "block" {
+				continue
+			}
+
+			// Add subsystems that need to be blacklisted here
```

Comment:
I'd add in code to support blacklisting now. The list might be empty, but it will then be easier to support future blacklisting, but also it will allow this code to be tested.

Speaking of tests, can you add some? If unit tests aren't possible/realistic, can you create a set of tests in https://github.com/clearcontainers/tests somewhere to exercise this new functionality?

Reply:
@jodh-intel See https://github.com/clearcontainers/tests/issues/794.

Reply:
I have added in the code to support blacklisting.